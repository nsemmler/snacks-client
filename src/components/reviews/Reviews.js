import React from 'react'
import {Container} from 'reactstrap';
import Review from '../../containers/Review'

const Reviews = ({props}) => {

  return (<Container id="reviews">
    {props.map(review => <Review key={review.id} review={review}/>)}
  </Container>)
}

export default Reviews
