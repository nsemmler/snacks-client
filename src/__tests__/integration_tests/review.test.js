import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Reviews from '../../components/reviews/Reviews'

const Provider = require('react-redux').Provider
const createStore = require('redux').createStore
const reducers = require('../../state/reducers')

describe('Reviews Component', () => {
  test('renders consistently', () => {
    let store = createStore(() => {})

    const reviews = [
      {
        id: 1,
        name: "Pork Rinds",
        description: "Mauris lacinia sapien quis libero. Nam dui. Proin leo odio, porttitor id, consequat in, consequat ut, nulla. Sed accumsan felis.",
        price: 8,
        img: "https://az808821.vo.msecnd.net/content/images/thumbs/0000398_salt-pepper-pork-rinds-2-oz_560.jpeg"
      },
      {
        id: 2,
        name: "Soup - Campbells Beef Noodle",
        description: "Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Etiam vel augue. Vestibulum rutrum rutrum neque.",
        price: 26,
        img: "https://images-na.ssl-images-amazon.com/images/I/71MavWF1P9L._SY550_.jpg"
      }
    ]

    const component = renderer.create(
      <Provider store={ store } >
        <MemoryRouter>
          <Reviews props={ reviews } />
        </MemoryRouter>
      </Provider>
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
