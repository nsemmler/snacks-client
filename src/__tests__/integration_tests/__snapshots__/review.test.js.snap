// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reviews Component renders consistently 1`] = `
<div
  className="container"
  id="reviews"
>
  <div
    className="container"
    id="review-block"
  >
    <div>
      <div
        className="row"
      >
        <h4
          id="review-title"
        />
      </div>
      <div
        className="row"
        id="review-block-rate"
      >
        <ul
          className="rc-rate"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          role="radiogroup"
          style={Object {}}
          tabIndex={0}
        >
          <li
            aria-checked="false"
            aria-posinset={1}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={2}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={3}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={4}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={5}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
        </ul>
      </div>
      <div
        className="row"
      />
      <br />
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-outline-warning btn-sm"
          onClick={[Function]}
          type="button"
        >
          Edit Review
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
    id="review-block"
  >
    <div>
      <div
        className="row"
      >
        <h4
          id="review-title"
        />
      </div>
      <div
        className="row"
        id="review-block-rate"
      >
        <ul
          className="rc-rate"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          role="radiogroup"
          style={Object {}}
          tabIndex={0}
        >
          <li
            aria-checked="false"
            aria-posinset={1}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={2}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={3}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={4}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
          <li
            aria-checked="false"
            aria-posinset={5}
            aria-setsize={5}
            className="rc-rate-star rc-rate-star-zero"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseMove={[Function]}
            role="radio"
            tabIndex={0}
          >
            <div
              className="rc-rate-star-first"
            >
              ★
            </div>
            <div
              className="rc-rate-star-second"
            >
              ★
            </div>
          </li>
        </ul>
      </div>
      <div
        className="row"
      />
      <br />
      <div
        className="btn-group"
        role="group"
      >
        <button
          className="btn btn-outline-warning btn-sm"
          onClick={[Function]}
          type="button"
        >
          Edit Review
        </button>
      </div>
    </div>
  </div>
</div>
`;
